nshlib/nsh_timcmds.c

```diff
@@ -595,8 +595,15 @@ int cmd_watch(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
 
   for (i = 0; i < count; i++)
     {
+      FAR char *buffer = lib_get_tempbuffer(LINE_MAX);
```

Comment:
why not get at line 595 & put at line 615 ?

Reply:
because line 613 call sleep, which may sleep for a long time.